src/Ldap/LdapData.php

```diff
@@ -0,0 +1,146 @@
+
+    /**
+     * Create a new LdapData object from an array of LDAP data.
+     *
+     * See https://confluence.cornell.edu/display/IDM/Attributes
+     *
+     * @param  array  $data  The LDAP data array, typically from a search result.
+     * @param  bool  $withLdapData  Whether to include the LDAP data in the returned object.
+     */
+    public static function make(array $data, bool $withLdapData = true): ?LdapData
+    {
+        // Use preferred first name if it is not null, otherwise use givenName.
+        $firstName = ($data['cornelleduprefgivenname'] ?? null) ?: ($data['givenName'] ?? null);
+
+        // Use preferred last name if it is not null, otherwise use sn.
+        $lastName = ($data['cornelleduprefsn'] ?? null) ?: ($data['sn'] ?? null);
+
+        // Affiliations can be a string, an array, or not set
+        $affiliationCollection = collect(($data['cornelleduaffiliation'] ?? null) ?: null);
+        $affiliations = $affiliationCollection->toArray();
+
+        // Use the defined primary affiliation or the first affiliation in the collection.
+        $primaryAffiliation = ($data['cornelleduprimaryaffiliation'] ?? null) ?: $affiliationCollection->shift() ?? '';
+
+        // Secondary affiliation is the first affiliation that is not the primary affiliation.
+        $secondaryAffiliation = $affiliationCollection->reject($primaryAffiliation)->first() ?? '';
+
+        // Process previousNetids: always convert the comma-separated string to an array.
+        $previousNetIds = collect(explode(',', $data['cornelledupreviousnetids'] ?? ''))
+            ->map(fn ($id) => trim($id))->filter()->all();
+
+        // Same for previousEmplids
+        $previousEmplids = collect(explode(',', $data['cornelledupreviousemplids'] ?? ''))
+            ->map(fn ($id) => trim($id))->filter()->all();
+
+        // User may have exercised FERPA right to suppress name.
+        if (empty($firstName) && empty($lastName)) {
+            $firstName = 'Cornell';
+            $lastName = $primaryAffiliation === 'student' ? 'Student' : 'User';
+        }
+
+        // Format an array to match the old LDAP::data function
+        $ldapData = [
+            'emplid' => $data['cornelleduemplid'] ?? '',
+            'firstname' => $firstName ?? '',
+            'lastname' => $lastName ?? '',
+            'name' => trim("$firstName $lastName"),
+            'email' => $data['mail'] ?? '',
+            'campusphone' => $data['cornelleducampusphone'] ?? '',
+            'netid' => $data['uid'],
+            'deptname' => $data['cornelledudeptname1'] ?? '',
+            'primaryaffiliation' => $primaryAffiliation,
+            'secondaryaffiliation' => $secondaryAffiliation,
+            'wrkngtitle' => $data['cornelleduwrkngtitle1'] ?? '',
+            'affiliations' => $affiliations,
+            'previousnetids' => $data['cornelledupreviousnetids'] ?? null,
+            'previousemplids' => $data['cornelledupreviousemplids'] ?? null,
+        ];
```

Comment:
Do we really need to return all these data from LDAP? Most of the time we just need to get a name and netid. I think the BindID must have special permissions to return some of these data (like emplid) and usually we do not have and do not need those permissions.